DSharpPlus.Commands/CommandsExtension.cs

```diff
@@ -108,6 +103,23 @@ public event AsyncEventHandler<CommandsExtension, CommandErroredEventArgs> Comma
 
     internal AsyncEvent<CommandsExtension, CommandErroredEventArgs> commandErrored;
 
+    /// <summary>
+    /// Executed before commands are finalized into a read-only state.
+    /// </summary>
+    /// <remarks>
+    /// Apply any mass-mutations to the commands or command parameters here.
+    /// </remarks>
+    public event AsyncEventHandler<
+        CommandsExtension,
+        ConfigureCommandsEventArgs
+    > ConfiguringCommands
+    {
+        add => this.configuringCommands.Register(value);
+        remove => this.configuringCommands.Unregister(value);
+    }
+
+    private AsyncEvent<CommandsExtension, ConfigureCommandsEventArgs> configuringCommands;

```

Comment:
why are we doing this instead of new dispatch?

Reply:
Because I haven't learned the new system and that was merged after I started this PR